scene/gui/color_picker.h

```diff
@@ -73,38 +89,52 @@ class ColorPicker : public BoxContainer {
 		SHAPE_MAX
 	};
 
+	static const int SLIDER_COUNT = 4;
+
 private:
 	static Ref<Shader> wheel_shader;
 	static Ref<Shader> circle_shader;
 	static Ref<Shader> circle_ok_color_shader;
 	static List<Color> preset_cache;
 
+	int current_slider_count = SLIDER_COUNT;
+
+	bool slider_theme_modified = true;
+
+	Vector<ColorMode *> modes;
+
 	Control *screen = nullptr;
-	Control *uv_edit = memnew(Control);
-	Control *w_edit = memnew(Control);
-	AspectRatioContainer *wheel_edit = memnew(AspectRatioContainer);
-	MarginContainer *wheel_margin = memnew(MarginContainer);
+	Control *uv_edit = nullptr;
+	Control *w_edit = nullptr;
+	AspectRatioContainer *wheel_edit = nullptr;
+	MarginContainer *wheel_margin = nullptr;
 	Ref<ShaderMaterial> wheel_mat;
 	Ref<ShaderMaterial> circle_mat;
-	Control *wheel = memnew(Control);
-	Control *wheel_uv = memnew(Control);
-	TextureRect *sample = memnew(TextureRect);
-	GridContainer *preset_container = memnew(GridContainer);
-	HSeparator *preset_separator = memnew(HSeparator);
-	Button *btn_add_preset = memnew(Button);
-	Button *btn_pick = memnew(Button);
-	CheckButton *btn_hsv = memnew(CheckButton);
-	CheckButton *btn_raw = memnew(CheckButton);
-	HSlider *scroll[4];
-	SpinBox *values[4];
-	Label *labels[4];
-	Button *text_type = memnew(Button);
-	LineEdit *c_text = memnew(LineEdit);
+	Control *wheel = nullptr;
+	Control *wheel_uv = nullptr;
+	TextureRect *sample = nullptr;
+	GridContainer *preset_container = nullptr;
+	HSeparator *preset_separator = nullptr;
+	Button *btn_add_preset = nullptr;
+	Button *btn_pick = nullptr;
+
+	OptionButton *mode_option_button = nullptr;
+
+	HSlider *sliders[SLIDER_COUNT];
+	SpinBox *values[SLIDER_COUNT];
+	Label *labels[SLIDER_COUNT];
+	Button *text_type = nullptr;
+	LineEdit *c_text = nullptr;
+
+	HSlider *alpha_slider = nullptr;
+	SpinBox *alpha_value = nullptr;
+	Label *alpha_label = nullptr;
 
 	bool edit_alpha = true;
 	Size2i ms;
 	bool text_is_constructor = false;
-	PickerShapeType picker_type = SHAPE_HSV_WHEEL;
+	PickerShapeType picker_type;
+	ColorModeType mode_type;
```

Comment:
```suggestion
	PickerShapeType current_shape;
	ColorModeType current_mode;
```
These would be better names.
Also they should be initialized to some default value (enums are basically ints).